mm2src/mm2_eth/src/eip712_encode.rs

```diff
@@ -144,7 +144,7 @@ fn encode_address(value: &Json, field_name: Option<&str>) -> Result<Vec<u8>> {
         .ok_or_else(|| expected_type_error("address", value, field_name))?;
     // "0x" - 2 chars, 40 chars are 20 hexed bytes.
     if string.len() != 42 {
-        return Err(decode_error("Expected 0x-prefixed address (20 bytes)", field_name))?;
+        Err(decode_error("Expected 0x-prefixed address (20 bytes)", field_name))?;
```

Comment:
let's bring the return back and not use `?` (replace it with `into` or something).
`Err()?` are return statements that look like expressions and hard to recognize.